pkg/hubble/metrics/api/api.go

```diff
@@ -63,18 +64,25 @@ type Handler interface {
 
 	// ProcessFlow must processes a flow event and perform metrics
 	// accounting
-	ProcessFlow(ctx context.Context, flow *pb.Flow)
+	ProcessFlow(ctx context.Context, flow *pb.Flow) error
 
 	// Status returns the configuration status of the metric handler
 	Status() string
 }
 
 // ProcessFlow processes a flow by calling ProcessFlow it on to all enabled
 // metric handlers
-func (h Handlers) ProcessFlow(ctx context.Context, flow *pb.Flow) {
+func (h Handlers) ProcessFlow(ctx context.Context, flow *pb.Flow) error {
+	var processingErr error
 	for _, mh := range h {
-		mh.ProcessFlow(ctx, flow)
+		err := mh.ProcessFlow(ctx, flow)
+		if err != nil {
+			// Continue running the remaining metrics handlers, since one failing
+			// shouldn't impact the other metrics handlers.
+			processingErr = multierr.Append(processingErr, err)
+		}
```

Comment:
nit: we don't necessarily need to check explicitly for err, as `multierr.Append` support either (or both) arguments to be `nil`

```suggestion
		err := mh.ProcessFlow(ctx, flow)
		processingErr = multierr.Append(processingErr, err)
		// Continue running the remaining metrics handlers, since one failing
		// shouldn't impact the other metrics handlers.
```

Reply:
Fixed